File: 搜索/深搜/leetcode559/559.py
"""
# Definition for a Node.
class Node(object):
    def __init__(self, val, children):
        self.val = val
        self.children = children
"""
class Solution(object):
    def maxDepth(self, root):
        """
        :type root: Node
        :rtype: int
        """
        if root == None:
            return 0
        self.result = 0
        self.dfs(root, 1)
        return self.result
    
    def dfs(self, root, depth):
        self.result = max(self.result, depth)
        for child in root.children:
            self.dfs(child, depth + 1)

File: 搜索/深搜/leetcode559/559.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/
class Solution {
public:
    int maxDepth(Node* root) {
        if (!root)
            return 0;
        if (root->children.size() == 0)
            return 1;
        int max_result = 0;
        for (int i = 0; i < root->children.size(); i++) {
            int result = maxDepth(root->children[i]);
            if (result > max_result)
                max_result = result;
        }
        return max_result + 1;  
    }
};